### PR TITLE
Creating translator without default category. 

### PR DESCRIPTION
**Pull request by darkdef**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ✔️

For usage without fake Category on create instance (for example - in modules)
see:
https://github.com/yii-extension/user-translations/blob/edc30bcc269728ee9b7ba2e5777cbc2c2fc943c5/config/common/yiisoft-translator.php#L53

### CODE REVIEW

File: src/Translator.php [hunk 1]
```diff
@@ -14,7 +14,7 @@
  */
 class Translator implements TranslatorInterface
 {
-    private string $defaultCategory;
+    private string $defaultCategory = '';
```
Comment (samdark) [inline src/Translator.php:17]: I wonder if we need default category at all. There was no ability to skip category in Yii 2. The downside if that in some cases you have to repeat yourself. The pro is that message extraction and overall logic becomes a bit simpler.

File: src/Translator.php [hunk 2]
```diff
@@ -24,23 +24,24 @@ class Translator implements TranslatorInterface
...
-     * @param CategorySource $defaultCategory Default category to use if category is not specified explicitly.
      * @param string $locale Default locale to use if locale is not specified explicitly.
+     * @param CategorySource|null $defaultCategory Default category to use if category is not specified explicitly, or null for use without default category
      * @param string|null $fallbackLocale Locale to use if message for the locale specified was not found. Null for none.
      * @param EventDispatcherInterface|null $eventDispatcher Event dispatcher for translation events. Null for none.
      */
     public function __construct(
-        CategorySource $defaultCategory,
         string $locale,
+        ?CategorySource $defaultCategory = null,
```
Comment (samdark) [inline src/Translator.php:34]:
```suggestion
        string $defaultCategory = 'app',
```

**Review by samdark — CHANGES_REQUESTED** [review]

File: src/Translator.php (hunk 2 above)
Comment (darkdef) [reply src/Translator.php:34]: Or without categorySource and usage defaultCategoryName, or with CategorySource(name, reader, formatter etc)

Comment (samdark) [reply src/Translator.php:34]: Or we can remove it altogether.

Comment (darkdef) [reply src/Translator.php]: yep

**Review by samdark — APPROVED** [review]

**Pull request merged**